lib/renderer.js

```diff
@@ -3,7 +3,8 @@
 const marked = require('marked');
 const { encodeURL, slugize, stripHTML, url_for, isExternalLink } = require('hexo-util');
 const MarkedRenderer = marked.Renderer;
-const { parse } = require('url');
+const { sep } = require('path');
+let hexo;
```

Comment:
IMHO it is not the best approach here.

Reply:
I agree, any suggestion?

Reply:
What about pass `hexo` through `constructor`?

And we can then use `new Renderer(this)` later.